types/inquirer/index.d.ts

```diff
@@ -9,21 +9,23 @@
 //                 Justin Rockwood <https://github.com/jrockwood>
 //                 Keith Kelly <https://github.com/kwkelly>
 //                 Richard Lea <https://github.com/chigix>
+//                 Jed Mao <https://github.com/jedmao>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.3
 
-// TypeScript Version: 2.8
-
-import through = require('through');
+import { ThroughStream } from 'through';
 import { Observable } from 'rxjs';
-import * as readline from 'readline';
+import { Interface as ReadlineInterface } from 'readline';
 
 declare namespace inquirer {
     type Prompts = { [name: string]: prompts.Base };
-    type ChoiceType = string | objects.ChoiceOption | objects.Separator;
-    type Questions<T = Answers> =
-        | Question<T>
-        | ReadonlyArray<Question<T>>
-        | Observable<Question<T>>;
+    type ChoiceType<A> = string | objects.ChoiceOption<A> | objects.Separator;
+    type Questions<A extends Answers = Answers> =
```

Comment:
Why not type it as `Questions<A extends Answers>` directly?

Reply:
Just wanted to provide a default so you can say `Questions` w/o having to write `Questions<A>`, but I see now how that doesn't make a lot of sense, because nobody would have empty answers, right? Is that your logic?

Reply:
Anyway, the default of `= Answers` was pre-existing, so maybe we'd do well to leave it alone?

Reply:
Using `Answers` as generic parameter by default makes it easier do write "quick'n'dirty" code - it might also be helpful if you extend an untyped module which asks questions using inquirer.

You might want to fetch the answers even if you don't really know what answers you might expect.

Reply:
I got it perfectly now. Thanks sincerely @jedmao @manuth